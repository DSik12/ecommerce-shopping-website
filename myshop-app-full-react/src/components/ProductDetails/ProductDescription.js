import React from 'react'
import './ProductDetails.scss';
import UpdateQuantity from '../shared/Cart/UpdateQuantity';

const Description = (props) => {
    return (
        <ul>
            <div id="description">
                <div className="postSection">
                    <h1 id="productName" >{props.productItems.productName}</h1>
                    <p>
                        {props.productItems.productDescription}
                    </p>
                    <div className="productPrice" id="productPrice"><h3>
                        Rs. {props.productItems.price}</h3>
                    </div>
                    < UpdateQuantity product={props.productItems} />
                </div>
            </div>
        </ul>
    )
}

export default Description;