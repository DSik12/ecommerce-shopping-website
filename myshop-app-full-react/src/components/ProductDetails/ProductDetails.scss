@import "../shared/ScssBreakpoint/Breakpoint.scss";
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: "Open Sans" !important;
}
@mixin slideshow-img-mixin($width, $height) {
  img {
    width: $width;
    height: $height;
  }
}
@mixin slideshow-thumbnail-mixin($display) {
  display: $display;
}
@mixin button-mixin($float, $margin-left, $margin-right, $padding) {
  float: $float;
  margin-left: $margin-left;
  margin-right: $margin-right;
  padding: $padding;
}
.sliderProductDetails {
  @include slideshow-img-mixin(300px, 300px);
  margin-left: 29px;
}
#thumbnailWrapper {
  @include slideshow-thumbnail-mixin(none);
}
#productName {
  margin-top: 50px;
}
.prevBtn {
  @include button-mixin(left, 30px, 15px, 5px);
}
.nextBtn { /* button */
  @include button-mixin(left, 265px, 0px, 5px);
}
#description { /* description */
  padding: 10px;
  margin: 30px 0px 30px 10px;
}
.productPrice { /* price margin */
  margin: 10px 5px;
}
.sliderProductDetails {
  img {
    margin: 120px 0px 0px 10px;
  }
}
@media only screen and (min-width: $tablet-portrait) {
  .sliderProductDetails { /* image size of slide show tiles */
    @include slideshow-img-mixin(600px, 600px);
  }
  #thumbnailWrapper { /* makes thumbnails flex and displays it back */
    @include slideshow-thumbnail-mixin(flex);
    float: left;
    margin-left: 65px;
  }

  .thumbnailImg { /* thumbnail size and margin */
    img {
      width: 130px;
      margin: 2px;
    }
  }
  .prevBtn {
    @include button-mixin(left, 60px, 15px, 5px);
  }
  .nextBtn { /* button */
    @include button-mixin(left, 500px, 0px, 5px);
  }
  #description { /* description */
    margin: 120px 0px 30px 10px;
  }
}
@media only screen and (min-width: $laptop) {
  .container { /* text and big image flex */
    display: flex;
    border-radius: 15px;
    margin: 130px 0px 30px;
  }
}