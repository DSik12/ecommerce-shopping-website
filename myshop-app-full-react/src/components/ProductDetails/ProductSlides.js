import React, { useState } from 'react';

const SlideshowProducts = (props) => {
    const [imageIndex, setIndex] = useState(0);

    const handleClickImg = (imgIndexOnClick) => {
        setIndex(prevImageIndex => prevImageIndex = imgIndexOnClick);
    }

    const handlePrevBtn = () => {
        if (imageIndex <= 0) {
            setIndex(prevImageIndex => prevImageIndex = 3);
        } else {
            setIndex(prevImageIndex => prevImageIndex - 1);
        }
    }

    const handleNextBtn = () => {
        if (imageIndex >= props.items.imageURLs.length - 1) {
            setIndex(prevImageIndex => prevImageIndex = 0);
        } else {
            setIndex(nextIndex => nextIndex + 1);
        }
    }

    return (
        <div>
            <div className='sliderProductDetails' id={`image${props.items.id}`}>
                <img src={props.items.imageURLs[imageIndex].url} alt={props.items.imageURLs[imageIndex].altText} />
            </div>
            <div>
                <button className="prevBtn" onClick={handlePrevBtn}>
                    ❮
                </button>
                <button className="nextBtn" onClick={handleNextBtn}>
                    ❯
                </button>
            </div>
            <div className="thumbnailWrapper" id="thumbnailWrapper">
                <span className="thumbnailImg" >
                    <img src={props.items.imageURLs[0].thumbnail} alt={props.items.imageURLs[0].altText} onClick={() => handleClickImg(0)} />
                </span>
                <span className="thumbnailImg" >
                    <img src={props.items.imageURLs[1].thumbnail} alt={props.items.imageURLs[1].altText} onClick={() => handleClickImg(1)} />
                </span>
                <span className="thumbnailImg" >
                    <img src={props.items.imageURLs[2].thumbnail} alt={props.items.imageURLs[2].altText} onClick={() => handleClickImg(2)} />
                </span>
                <span className="thumbnailImg" >
                    <img src={props.items.imageURLs[3].thumbnail} alt={props.items.imageURLs[3].altText} onClick={() => handleClickImg(3)} />
                </span>
            </div>
        </div>
    )
}

export default SlideshowProducts; 