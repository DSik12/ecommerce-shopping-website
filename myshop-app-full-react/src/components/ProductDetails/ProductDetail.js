import Description from './ProductDescription';
import './ProductDetails.scss';
import SlideshowProducts from './ProductSlides';
import useFetch from '../../config/useFetch'
import { useParams } from 'react-router-dom'
import fetchUrls from '../../config/config.json'

const ProductDetails = () => {
    const myParams = useParams();
    const products = useFetch(`${fetchUrls.productApi}?id=${myParams.id}`)
    let productItem = null;
    if (products.apiData !== null) {
        productItem = products.apiData.map((item) => {
            return (<div className="container" key={item} >
                <div className="slideshow-container">
                    <SlideshowProducts items={item} />
                </div>
                <Description productItems={item} />
            </div>)
        });
    }
    return (
        <div className="slideshow-container" id="slideshow-container">
            {productItem && productItem.length > 0 ?
                productItem
                :
                <div className='text-center'>
                    <h1>No Products Found</h1>
                </div>
            }
        </div>
    )
}


export default ProductDetails;



