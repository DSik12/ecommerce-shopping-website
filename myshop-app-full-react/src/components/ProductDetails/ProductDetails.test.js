import React from "react";
import ReactDOM from "react-dom";
import ProductDetails from "./ProductDetail";
import renderer from "react-test-renderer";

describe("Productdetails Component", () => {
  it("renders product details component successfully", () => {
    const divEl = document.createElement("div");
    ReactDOM.render(<ProductDetails />, divEl);
  });

  it("should have right component snapshot", () => {
    const tree = renderer.create(<ProductDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
