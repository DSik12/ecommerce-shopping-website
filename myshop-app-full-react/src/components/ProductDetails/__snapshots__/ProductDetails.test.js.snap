// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Productdetails Component should have right component snapshot 1`] = `
<div
  className="slideshow-container"
  id="slideshow-container"
>
  <div
    className="text-center"
  >
    <h1>
      No Products Found
    </h1>
  </div>
</div>
`;
